### PR TITLE
[RFR] Work so far

### PR DESCRIPTION
**Pull request by ThieryMichel**

### CODE REVIEW

**Comment by fzaninotto** [comment]
Wow, hard to review. Let's discuss this PR next time we meet.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+sudo: required
```
Comment (djhi) [inline .travis.yml]: I think we would have faster tests if running travis without sudo nor docker and using the postgres service.

File: lib/queries/batchParameter.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    return Object.keys(object)
+    .reduce((result, key) => {
+        return {
+            ...result,
+            [key + suffix]: object[key]
+        };
+    }, {});
+};
+
+export function merge(first, second) {
```
Comment (djhi) [inline lib/queries/batchParameter.js]: A simple Object.assign would work instead of this function

File: lib/queries/batchUpsert.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            fields
+        } = config;
+        const getValueSubQuery = valueSubQuery(fields);
+        const getParameter = batchParameter(fields);
+
+        if (Array.isArray(returnFields)) {
+            returnFields = returnFields.join(', ');
+        }
+
+        const setQuery = updatableFields.map(function (field) {
```
Comment (djhi) [inline lib/queries/batchUpsert.js]: arrow func ?

File: lib/queries/batchParameter.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import sanitizeParameter from './sanitizeParameter';
+import arrayToLitteral from '../utils/arrayToLitteral';
+
+export function addSuffix(object, suffix) {
+    return Object.keys(object)
+    .reduce((result, key) => {
```
Comment (djhi) [inline lib/queries/batchParameter.js]: You can use object notation here:
```
.reduce((result, key) => ({
    ...result,
    [key + suffix]: object[key]
}), {});

File: lib/queries/selectPage.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                }, `CASE ${sort}`);
+                sortQuery.unshift(`${specificSort} END ${(sortDir.toLowerCase() === 'asc' ? ' ASC' : ' DESC')}`);
+            } else if (returnFields.indexOf(sort) !== -1) {
+                sortQuery.unshift(`${sort} ${(sortDir.toLowerCase() === 'asc' ? 'ASC' : 'DESC')}`);
+            }
+        }
+        sql += ` ORDER BY ${sortQuery.join(', ')}`;
+
+        if (limit) {
+            sql += ` LIMIT $limit OFFSET $offset`;
```
Comment (djhi) [inline lib/queries/selectPage.js]: s/sql += ` LIMIT/`${sql} LIMIT`

File: lib/queries/upsertOne.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+            selectorFields,
+            updatableFields,
+            returnFields
+        } = config;
+
+        const parameters = sanitizeParameter(fields, entity);
+        const keys = Object.keys(parameters);
+        const values = valueSubQuery(keys, '');
+
+        const setQuery = keys.filter(key => updatableFields.indexOf(key) !== -1).map(function (field) {
```
Comment (djhi) [inline lib/queries/upsertOne.js]: arrow func everywhere ?
```
key
    .filter(...)
    .map(...)
```

more readable imho

File: lib/utils/combineLiterals.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+export default function combineLiterals(literals) {
+    return literals.reduce((result, literal) => {
+        return {
+            ...result,
+            ...Object.keys(literal).reduce((r, key) => {
```
Comment (djhi) [inline lib/utils/combineLiterals.js]: object notation for simpler return

File: lib/utils/configurable.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+export default function configurable(targetFunction, config) {
```
Comment (djhi) [inline lib/utils/configurable.js]: Why not creating a simple lib out of this ? :+1:

File: lib/utils/middleware.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+export default function init(entry, options) {
+    var handlers = [];
+
+    return {
```
Comment (djhi) [inline lib/utils/middleware.js]: you can use es6 funcs notation here
```
use(fn, target) {
},
execute(emptyResult) {
}
```

File: lib/utils/middleware.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+        use: function (fn, target) {
+            handlers.push({ fn, target });
+            return this;
+        },
+        execute: function execute(emptyResult) {
+            return handlers.reduce(function (result, { fn, target }) {
```
Comment (djhi) [inline lib/utils/middleware.js]: arrow func ?

**Comment by djhi** [comment]
outstanding work ! 

We should probably apply airbnb style everywhere like in our others projects but that might be done in another pr.

File: lib/queries/batchParameter.js (hunk 2 above)
Comment (jpetitcolas) [inline lib/queries/batchParameter.js]: Or simply spread operator.

File: lib/utils/middleware.js (hunk 9 above)
Comment (ThieryMichel) [inline lib/utils/middleware.js]: Not for use I need the `this`.

Comment (djhi) [inline lib/utils/middleware.js]: I didn't say arrow funcs

Comment (ThieryMichel) [inline lib/utils/middleware.js]: Ah? I thouht this worked only for class. Good to know.

File: .travis.yml (hunk 1 above)
Comment (ThieryMichel) [inline .travis.yml]: Not possible actually, there is no travis add-on yet for postgres 9.5 which I use, and need for the upsert: 
`ON CONFLICT UPDATE`

File: lib/utils/middleware.js (hunk 9 above)
Comment (djhi) [inline lib/utils/middleware.js]: Me toi, discovered it with https://egghead.io/lessons/javascript-redux-simplifying-the-arrow-functions

Comment (ThieryMichel) [inline lib/utils/middleware.js]: Merci pur le lien.
done

**Comment by ThieryMichel** [comment]
Review applied, need configurable.js 0.1.1 to be published.

**Comment by ThieryMichel** [comment]
Ready

File: README.md [hunk 11]
```diff
@@ -0,0 +1,203 @@
+#coPostgresQueries
+utility to generate and execute postgresql queries with ease.
+
+##pgClient
+Allow to cennect to postgres and execute query
```
Comment (Kmaschta) [inline README.md]: s/cennect/connect

**Pull request merged**